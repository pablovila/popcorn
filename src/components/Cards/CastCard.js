import React from "react";
import PropTypes from "prop-types";

import { IMAGE_BASE_URL } from "apis/tmdb/apiConfig";

const castCard = ({ person }) => {
  return (
    <div className="card">
      <div className="card-image">
        <figure className="image">
          {person.profile_path ? (
            <img src={`${IMAGE_BASE_URL}w185${person.profile_path}`} />
          ) : (
            <></>
          )}
        </figure>
      </div>
      <div className="card-content">
        <div className="content">
          <h5 className="title is-5">{person.name}</h5>
          <h6 className="subtitle is-6">{person.character}</h6>{" "}
        </div>
      </div>
    </div>
  );
};

castCard.propTypes = {
  person: PropTypes.object.isRequired
};

export default castCard;
