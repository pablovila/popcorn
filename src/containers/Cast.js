import React from "react";
import PropTypes from "prop-types";

import CastCard from "components/Cards/CastCard";

const cast = ({ cast }) => {
  return (
    <div className="columns is-multiline">
      {cast.slice(0, 12).map(person => (
        <div className="column is-2" key={person.id}>
          <CastCard person={person} />
        </div>
      ))}
    </div>
  );
};

cast.propTypes = {
  cast: PropTypes.array.isRequired
};

export default cast;
