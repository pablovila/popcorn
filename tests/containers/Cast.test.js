import React from "react";
import { shallow, mount } from "enzyme";

import Cast from "containers/Cast";
import CastCard from "components/Cards/CastCard";

const mockCast = [
  {
    id: 60073,
    name: "Brie Larson",
    character: "Carol Danvers / Vers / Captain Marvel"
  }
];

describe("test Cast container", () => {
  it("renders Cast container", () => {
    const cast = shallow(<Cast cast={mockCast} />);

    expect(cast.find("div.columns")).toHaveLength(1);
    expect(cast).toMatchSnapshot();
  });

  it("renders Cast children", () => {
    const cast = mount(<Cast cast={mockCast} />);

    expect(cast.find(CastCard)).toHaveLength(1);
    expect(cast).toMatchSnapshot();
  });
});
