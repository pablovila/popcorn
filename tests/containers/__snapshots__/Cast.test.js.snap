// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Cast container renders Cast children 1`] = `ReactWrapper {}`;

exports[`test Cast container renders Cast container 1`] = `ShallowWrapper {}`;
